### PR TITLE
cortx-33317

### PR DESCRIPTION
**Pull request by sonalk0209**
# Problem Statement
small Lib and one test change for data pod restart
# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: tests/ha/single_pod_restart/test_data_pod_restart.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def setup_class(cls):
         cls.host_worker_list = []
         cls.node_worker_list = []
         cls.ha_obj = HAK8s()
```
Comment (ApurwaMohite) [inline tests/ha/single_pod_restart/test_data_pod_restart.py:81]: Please change the class name as well.

File: tests/ha/single_pod_restart/test_data_pod_restart.py [hunk 2]
```diff
@@ -183,88 +174,89 @@ def teardown_method(self):
...
-        LOGGER.debug("Response: %s", resp)
-        assert_utils.assert_true(resp[0], resp)
-        LOGGER.info("Step 4: Services of pod are in online state")
-
-        LOGGER.info("Step 5: Perform WRITEs with variable object sizes. (0B - 512MB(VM)/5GB(HW))")
+        LOGGER.info("STEP 1: Perform WRITEs/READs/Verify with variable object sizes")
         users = self.mgnt_ops.create_account_users(nusers=1)
         self.test_prefix = 'test-34072'
-        self.s3_clean.update(users)
+        self.s3_clean = users
```
Comment (ApurwaMohite) [inline tests/ha/single_pod_restart/test_data_pod_restart.py:187]: Update was also correct right?

**Review by ApurwaMohite — APPROVED** [review]

File: tests/ha/single_pod_restart/test_data_pod_restart.py (hunk 2 above)
Comment (sonalk0209) [reply tests/ha/single_pod_restart/test_data_pod_restart.py:187]: yeah

File: libs/ha/ha_common_libs_k8s.py [hunk 3]
```diff
@@ -862,14 +862,14 @@ def poll_cluster_status(self, pod_obj, timeout=1200):         # default 20mins t
...
-    @staticmethod
-    def restore_pod(pod_obj, restore_method, restore_params: dict = None):
+    def restore_pod(self, pod_obj, restore_method, restore_params: dict = None, clstr_status=False):
         """
         Helper function to restore pod based on way_to_restore
         :param pod_obj: Object of master node
         :param restore_method: Restore method to be used depending on shutdown method
         ("scale_replicas", "k8s", "helm")
         :param restore_params: Dict which has parameters required to restore pods
+        :clstr_status: Flag to check cluster status fter pod restored
```
Comment (ankita-93) [inline libs/ha/ha_common_libs_k8s.py:872]: after *

File: tests/ha/single_pod_restart/test_data_pod_restart.py [hunk 4]
```diff
@@ -183,88 +174,89 @@ def teardown_method(self):
...
                                                     skipcleanup=True)
         assert_utils.assert_true(resp[0], resp[1])
-        LOGGER.info("Step 5: Performed WRITEs with variable sizes objects.")
+        LOGGER.info("Step 3: Performed READs/Verify on data written in healthy cluster.")
 
-        LOGGER.info("Step 6: Starting pod again by creating deployment using K8s command")
+        LOGGER.info("Step 4: Perform WRITEs/READs/Verify with variable object sizes.")
+        if CMN_CFG["dtm0_disabled"]:
+            self.test_prefix_deg = 'test-34072-deg'
+            resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
```
Comment (ankita-93) [inline tests/ha/single_pod_restart/test_data_pod_restart.py:218]: can we skip re-installion of s3bench

File: tests/ha/single_pod_restart/test_data_pod_restart.py [hunk 5]
```diff
@@ -183,88 +174,89 @@ def teardown_method(self):
...
+                                                        log_prefix=self.test_prefix_deg,
+                                                        skipcleanup=True)
+        else:
+            resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
```
Comment (ankita-93) [inline tests/ha/single_pod_restart/test_data_pod_restart.py:222]: same. no need to install s3bench again

File: tests/ha/single_pod_restart/test_data_pod_restart.py [hunk 6]
```diff
@@ -183,88 +174,89 @@ def teardown_method(self):
...
+                                                        skipwrite=True,
+                                                        skipcleanup=True)
+            assert_utils.assert_true(resp[0], resp[1])
+            LOGGER.info("Step 6: Successfully run READ/Verify on data written in degraded cluster "
+                        "with new buckets and objects.")
+        LOGGER.info("Step 6: Perform READs and verify DI on the written data with buckets created "
+                    "in healthy cluster.")
         resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
                                                     log_prefix=self.test_prefix, skipwrite=True,
                                                     skipcleanup=True)
```
Comment (ankita-93) [inline tests/ha/single_pod_restart/test_data_pod_restart.py:255]: sebench installion = False

File: tests/ha/single_pod_restart/test_data_pod_restart.py (hunk 4 above)
Comment (sonalk0209) [reply tests/ha/single_pod_restart/test_data_pod_restart.py:218]: yeah makes sense.

**Comment by sonalk0209** [comment]
[pr1733.log](https://github.com/Seagate/cortx-test/files/9070811/pr1733.log)
Collections logs

**Review by ankita-93 — APPROVED** [review]

**Review by ravindrachoudhari — APPROVED** [review]

**Pull request merged**